src/test/java/com/googleinterns/zoomtube/servlets/TranscriptServletTest.java

```diff
@@ -0,0 +1,254 @@
+    when(request.getParameter(TranscriptServlet.PARAM_VIDEO_ID)).thenReturn(LONG_VIDEO_ID);
+    when(request.getParameter(TranscriptServlet.PARAM_LECTURE_ID)).thenReturn(LECTURE_ID_A);
+
+    servlet.doPost(request, response);
+    servlet.doGet(request, response);
+
+    String actualJson = lectureTranscript.toString();
+    List<TranscriptLine> expectedArrayList = extractJson(LONG_VIDEO_JSON);
+    List<TranscriptLine> actualJsonArray = extractJson(actualJson);
+    assertThat(actualJsonArray).isEqualTo(expectedArrayList);
+  }
+
+  @Test
+  public void doGet_returnsLectureForLongVideoFromDatastore() throws ServletException, IOException {
+    putJsonInDatastore(LONG_VIDEO_JSON, LECTURE_ID_A);
+    when(request.getParameter(TranscriptServlet.PARAM_LECTURE_ID)).thenReturn(LECTURE_ID_A);
+
+    servlet.doGet(request, response);
+
+    String actualJson = lectureTranscript.toString();
+    List<TranscriptLine> expectedArrayList = extractJson(LONG_VIDEO_JSON);
+    List<TranscriptLine> actualJsonArray = extractJson(actualJson);
+    assertThat(actualJsonArray.size()).isEqualTo(expectedArrayList.size());
+  }
+
+  @Test
+  public void doPost_persistDataInDatastoreForLongVideo() throws ServletException, IOException {
+    when(request.getParameter(TranscriptServlet.PARAM_VIDEO_ID)).thenReturn(LONG_VIDEO_ID);
+    when(request.getParameter(TranscriptServlet.PARAM_LECTURE_ID)).thenReturn(LECTURE_ID_C);
+
+    servlet.doPost(request, response);
+
+    int actualQueryCount = entitiesInDatastoreCount(LECTURE_ID_C);
+    int expectedQueryCount = (extractJson(LONG_VIDEO_JSON)).size();
+    assertThat(actualQueryCount).isEqualTo(expectedQueryCount);
+  }
+
+  @Test
+  public void doGet_onlyOtherLecturesInDatastore_GetNoLectures()
+      throws ServletException, IOException {
+    putJsonInDatastore(SHORT_VIDEO_JSON, LECTURE_ID_B);
+    putJsonInDatastore(LONG_VIDEO_JSON, LECTURE_ID_A);
+    when(request.getParameter(TranscriptServlet.PARAM_LECTURE_ID)).thenReturn(LECTURE_ID_C);
+
+    servlet.doGet(request, response);
+
+    String actualJson = lectureTranscript.toString();
+    List<TranscriptLine> actualJsonArray = extractJson(actualJson);
+    assertThat(actualJsonArray.size()).isEqualTo(0);
+  }
+
+  @Test
+  public void doGet_twoLecturesInDatastore_returnsOneLecture()
+      throws ServletException, IOException {
+    putJsonInDatastore(SHORT_VIDEO_JSON, LECTURE_ID_B);
+    putJsonInDatastore(LONG_VIDEO_JSON, LECTURE_ID_A);
+    when(request.getParameter(TranscriptServlet.PARAM_LECTURE_ID)).thenReturn(LECTURE_ID_A);
+
+    servlet.doGet(request, response);
+
+    String actualJson = lectureTranscript.toString();
+    List<TranscriptLine> expectedArrayList = extractJson(LONG_VIDEO_JSON);
+    List<TranscriptLine> actualJsonArray = extractJson(actualJson);
+    assertThat(actualJsonArray.size()).isEqualTo(expectedArrayList.size());
+  }
+
+  private List<TranscriptLine> extractJson(String json) {
```

Comment:
Actually I know I mentioned calling it extractJson. I think a better name would be getTranscriptLines or just transcriptLines. This indicates what you are obtaining.